File: python/1071_number_spiral.py
def get_value(row: int, col: int) -> int:
    k = max(row, col)
    # Diagonal value
    d = k * (k - 1) + 1
    # Add or subtract along the axis
    if k % 2 == 0:
        val = d - (col - row)
    else:
        val = d + (col - row)
    return val


n = int(input())

for i in range(n):
    row, col = [int(x) for x in input().split()]
    print(get_value(row, col))

File: src/1071_number_spiral.rs
use std::io::{self, BufWriter, Write};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let input = io::stdin();
    let mut lines = input.lines();
    // Flush all at once
    let mut output = BufWriter::new(io::stdout().lock());

    lines.next();

    for line in lines {
        let line = line?;
        let (row, col) = line.split_once(' ').unwrap();
        let row = row.parse::<usize>()?;
        let col = col.parse::<usize>()?;

        let k = row.max(col);
        // Diagonal value
        let d = k * (k - 1) + 1;
        let value = if k % 2 == 0 {
            d - (col - row)
        } else {
            d + (col - row)
        };
        writeln!(output, "{value}")?;
    }
    Ok(())
}
